0048-schema-update-process/README.md

```diff
@@ -0,0 +1,391 @@
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this ZEP.  Describe why the change is important and the benefits to users. You
+can also optionally include links to [experience reports], [community slacks],
+or other references to show the community's interest in the ZEP.
+
+[experience reports]: https://go.dev/wiki/ExperienceReports
+[openssf slack]: https://openssf.slack.com/archives/C07AKUMBDMJ
+[kubernetes slack]: https://kubernetes.slack.com/archives/C03B6BJAUJ3
+-->
+
+There are several open issues requesting enhancements to the schema, but before Zarf introduces a new schema, there must be a plan to handle schema upgrades. The general theme of these changes is to make the ZarfPackageConfig schema more intuitive to use. 
+- [Refactor charts definition in zarf.yaml #2245](https://github.com/zarf-dev/zarf/issues/2245)
+- [Breaking Change: make components required by default #2059](https://github.com/zarf-dev/zarf/issues/2059)
+- [Use kstatus as the engine behind zarf tools wait-for and .wait.cluster #4077](https://github.com/zarf-dev/zarf/issues/4077)
+
+ZEP [0051-v1beta1-schema](https://github.com/zarf-dev/proposals/pull/52) provides the specifics for what will change in the next schema version of Zarf.
+
+### Goals
+
+<!--
+List the specific goals of the ZEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Provide clear guidelines for how Zarf package commands should behave when handling new or old schema versions.
+- Design a strategy for updating the codebase when a new schema is introduced. 
+- Introduce a command for users to upgrade the schema version of their Zarf package config.
+
+### Non-Goals
+
+<!--
+What is out of scope for this ZEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Define the next API version of the ZarfPackageConfig
+
+## Proposal
+
+<!--
+This is where you explain the specifics of the proposal. Provide enough detail
+for reviewers to clearly understand what you're proposing, but avoid including
+too many specifics like API designs or implementation details. Focus on the
+desired outcome and how success will be measured. The "Design Details" section
+below is for the real nitty-gritty.
+-->
+
+During Zarf's lifetime, it will introduce, deprecate, and drop support for ZarfPackageConfig API versions. Once a version is deprecated, users will still be able to perform all package operations such as create, publish, and deploy, but will receive warnings that they should upgrade. Zarf will drop support for an API version one year after it is deprecated. Once an API version is no longer supported, Zarf will error if a user tries to perform any zarf package operations with that API version. 
```

Comment:
I like the "support for one year" expectation here - it's farther out than we could reasonable expect to tie it to a specific release. Maybe we can have an SDPX tag for the expected deprecation time? Automating it is a bit of scope creep, but I like having the hook there already.

Reply:
Just for Zarf's own historicals it has used CLI version cutoffs for feature removals - https://github.com/zarf-dev/zarf/blob/a26516131a5df8dd2ddc93ec1f2e59bd959c971d/zarf.schema.json#L539 - not saying that is the right way but something to compare against.

Reply:
Yeah, I hadn't thought much about deployedPackages. They are used during inspect and remove. Added a line about this that extends deployed packages for another year. We have to keep them around for longer as we don't want to make it impossible for someone to remove an old deprecated package.